langchain/document_loaders/trello.py

```diff
@@ -0,0 +1,131 @@
+        include_comments: bool = True,
+        include_checklist: bool = True,
+        card_filter: str = "all",
+        extra_metadata: tuple[str, ...] = ("due_date", "labels", "list", "is_closed"),
+    ):
+        """Initialize Trello loader."""
+        self.api_key = api_key
+        self.api_token = api_token
+        self.board_name = board_name
+        self.include_card_name = include_card_name
+        self.include_comments = include_comments
+        self.include_checklist = include_checklist
+        self.extra_metadata = extra_metadata
+        self.card_filter = card_filter
+
+    def load(self) -> List[Document]:
+        """Loads all cards from the specified Trello board.
+        You can filter the cards, metadata and text included by using the optional parameters.
+
+         Returns:
+            A list of documents, one for each card in the board.
+        """
+
+        try:
+            from trello import TrelloClient  # type: ignore
+        except ImportError as ex:
+            raise ImportError(
+                "Could not import trello python package. "
+                "Please install it with `pip install py-trello`."
+            ) from ex
+        try:
+            from bs4 import BeautifulSoup  # type: ignore
+        except ImportError as ex:
+            raise ImportError(
+                "`beautifulsoup4` package not found, please run"
+                " `pip install beautifulsoup4`"
+            ) from ex
+
+        docs: List[Document] = []
+        client = TrelloClient(api_key=self.api_key, token=self.api_token)
+
+        # Find the board with the matching name
+        board = next(
+            (b for b in client.list_boards() if b.name == self.board_name), None
+        )
+        if not board:
+            raise ValueError(f"Board `{self.board_name}` not found.")
+
+        # Create a dictionary with the list IDs as keys and the list names as values
+        list_dict = {list_item.id: list_item.name for list_item in board.list_lists()}
+
+        # Get Cards on the board
+        cards = board.get_cards(card_filter=self.card_filter)
+        for card in cards:
+            text_content = ""
+            if self.include_card_name:
+                text_content = card.name + "\n"
+            description = card.description.strip()
+            if description:
+                text_content += BeautifulSoup(card.description, "lxml").get_text()
+
+            if self.include_checklist:
+                # Get all the checklit items on the card
+                items = []
+                for checklist in card.checklists:
+                    if checklist.items:
+                        items.extend(
+                            [
+                                f"{item['name']}:{item['state']}"
+                                for item in checklist.items
+                            ]
+                        )
+                        text_content += f"\n{checklist.name}\n" + "\n".join(items)
```

Comment:
guessing only new items should be part of the join? otherwise you'll be reading all the items added from a previous checklists

Reply:
Good catch! 
I saw that you did a little refactor, thanks!